packages/core-data/src/dynamic-actions.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Internal dependencies
+ */
+import type * as ET from './entity-types';
+
+export namespace DynamicActions {
+	type DeleteRecordsHttpQuery = Record< string, any >;
+
+	type SaveRecordOptions = {
+		throwOnError?: boolean;
+	};
+
+	type DeleteRecordOptions = {
+		throwOnError?: boolean;
+	};
+
+	/**
+	 * Save the site settings
+	 *
+	 * @param data    The site settings
+	 * @param options
+	 */
+	export declare function saveSite(
+		data: Partial< ET.Settings >,
+		options?: SaveRecordOptions
+	): Promise< void >;
+
+	/**
+	 * Save comment
+	 *
+	 * @param data    The comment data
+	 * @param options
+	 */
+	export declare function saveComment(
```

Comment:
I'm not sure it's a good idea to hardcode all these. Even if we listed them all (which I'm not sure we did) how can we guarantee they're up to date if we add a new entity in the future? Furthermore, some of those are actually dynamic, based on actual registered post types and taxonomies. We don't seem to consider those here, do we? With that in mind, we might want to make the type a bit broader. Knowing how tricky it has been historically to type custom Redux implementations, this can be a fun challenge. cc @jsnajdr for thoughts

Reply:
Agreed. Could we make these somehow dynamic based on `EntityRecord` or so? And then maybe make them generic so that anyone with custom post types could add them on top?

Reply:
Perhaps if we keep the types more open in a dictionary, and just provide some signatures like `save*` and `delete*` for actions and `get*` for selectors? See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#symbol-and-template-string-pattern-index-signatures

Reply:
Alright, that is going to test my TS skills. I am up for the challenge. 😄

Reply:
OK, I have done it. Mission accomplished 😄

Reply:
Nice, thanks for taking it 😅 I'll try to get you another review next week.